app/celery/letters_pdf_tasks.py

```diff
@@ -144,3 +152,42 @@ def letter_in_created_state(filename):
             notifications[0].status
         ))
     return False
+
+
+@notify_celery.task(name='process-virus-scan-passed')
+def process_virus_scan_passed(filename):
+    current_app.logger.info('Virus scan passed: {}'.format(filename))
+    move_scanned_pdf_to_letters_pdf_bucket(filename)
+    reference = get_reference_from_filename(filename)
+    updated_count = update_letter_pdf_status(reference, NOTIFICATION_CREATED)
+
+    if updated_count != 1:
+        raise Exception(
+            "There should only be one letter notification for each reference. Found {} notifications".format(
+                updated_count
+            )
+        )
+
+
+@notify_celery.task(name='process-virus-scan-failed')
+def process_virus_scan_failed(filename):
+    current_app.logger.info('Virus scan failed: {}'.format(filename))
+    delete_pdf_from_letters_scan_bucket(filename)
```

Comment:
i'd rather do this delete last - just so it doesn't run if something else unexpected happens?